### PR TITLE
fix storage

### PR DESCRIPTION
**Pull request by smnzhu**
Updates the execution data provider to track fulfilled height.

### CODE REVIEW

**Review by koko1123 — APPROVED** [review]

**Comment by peterargue** [comment]
@smnzhu can you add some context to the description on what this change is fixing?

**Review by peterargue — APPROVED** [review]

**Comment by smnzhu** [comment]
bors merge

**Comment by peterargue** [comment]
@smnzhu  Looks like there are some legit unit test failures

File: engine/execution/computation/execution_verification_test.go [hunk 1]
```diff
@@ -674,10 +672,10 @@ func executeBlockAndVerifyWithParameters(t *testing.T,
...
 
 	bservice := requesterunit.MockBlobService(blockstore.NewBlockstore(dssync.MutexWrap(datastore.NewMapDatastore())))
-	trackerStorage := new(mocktracker.Storage)
-	trackerStorage.On("Update", mock.Anything).Return(func(fn exedatatracker.UpdateFn) error {
-		return fn(func(uint64, ...cid.Cid) error { return nil })
-	})
+	trackerStorage := mocktracker.NewMockStorage()
+
+	// Assume SetFulfilledHeight succeeds unless otherwise indicadted in a test requriement
+	trackerStorage.On("SetFulfilledHeight", mock.Anything).Return(nil)
```
Comment (peterargue) [inline engine/execution/computation/execution_verification_test.go:678]: this line is also run in `mocktracker.NewMockStorage()` so you can probably skip it here

**Review by peterargue — APPROVED** [review]
thanks @Kay-Zee!

**Review by janezpodhostnik — APPROVED** [review]

**Review by pattyshack — APPROVED** [review]

**Comment by Kay-Zee** [comment]
bors merge

**Pull request merged**